### PR TITLE
bpo-40484: Document compiler flags under AST module

### PR DESCRIPTION
**Pull request by isidentical**
https://bugs.python.org/issue40484

### CODE REVIEW

File: Doc/library/ast.rst [hunk 1]
```diff
@@ -1748,6 +1748,34 @@ and classes for traversing abstract syntax trees:
       Added the *indent* option.
 
 
+.. _ast-compiler-flags:
+
+Compiler Flags
```
Comment (pablogsal) [inline Doc/library/ast.rst:1753]: I understand why you documented these here, but I am worried about the discovery of these flags. Should it be a cross-reference in the `compile` function docs? Are these even mentioned there?

Comment (isidentical) [reply Doc/library/ast.rst:1753]: IMHO this is the initial step to increase that `non-existent` discoverability. I believe we may need to re-write whole flags (and dont_inherit) documentation again for describing them supporting not only future features, but compiler flags in an another PR.

Comment (pablogsal) [reply Doc/library/ast.rst:1753]: Giving that this PR is rather small, I would suggest doing this in this PR as the decisions we take there may affect this work (for instance we may decide that these flags will be documented only on the `compile` function, rendering this work obsolete.

Comment (isidentical) [reply Doc/library/ast.rst:1753]: I made the required adjustments.

File: Doc/library/ast.rst [hunk 2]
```diff
@@ -1748,6 +1748,34 @@ and classes for traversing abstract syntax trees:
...
+--------------
+
+Following flags may passed to :func:`compile` in order to change
+affects on compilation of a program:
```
Comment (hauntsaninja) [inline Doc/library/ast.rst]:
```suggestion
The following flags may be passed to :func:`compile` in order to change
effects on compilation of a program:
```

File: Doc/library/functions.rst [hunk 3]
```diff
@@ -246,26 +246,24 @@ are always available.  They are listed here in alphabetical order.
...
-   The optional argument *flags* also controls whether the compiled source is
-   allowed to contain top-level ``await``, ``async for`` and ``async with``.
-   When the bit ``ast.PyCF_ALLOW_TOP_LEVEL_AWAIT`` is set, the return code
-   object has ``CO_COROUTINE`` set in ``co_code``, and can be interactively
-   executed via ``await eval(code_object)``.
+   The optional argument *flags* and *dont_inherit* controls which
+   :ref:`compiler options <ast-compiler-flags>` should be activated
+   and which :ref:`future features <future>` should be allowed. If neither
+   is present (or both are zero) the code is compiled with the same flags that
+   are effecting the code that is calling :func:`compile`. If the *flags*
```
Comment (hauntsaninja) [inline Doc/library/functions.rst]:
```suggestion
   affect the code that is calling :func:`compile`. If the *flags*
```

**Review by hauntsaninja — COMMENTED** [review]
Two small grammar nits! :-)

**Comment by isidentical** [comment]
> Two small grammar nits! :-)

Thanks @hauntsaninja!

File: Doc/library/ast.rst [hunk 4]
```diff
@@ -1748,6 +1748,34 @@ and classes for traversing abstract syntax trees:
       Added the *indent* option.
 
 
+.. _ast-compiler-flags:
+
+Compiler Flags
+--------------
+
+The following flags may be passed to :func:`compile` in order to change
+effects on compilation of a program:
```
Comment (pablogsal) [inline Doc/library/ast.rst]:
```suggestion
The following flags may be passed to :func:`compile` in order to change
effects on the compilation of a program:
```

File: Doc/library/ast.rst [hunk 5]
```diff
@@ -1748,6 +1748,34 @@ and classes for traversing abstract syntax trees:
...
+
+   Enables support for top-level ``await``, ``async for``, ``async with``
+   and async comprehensions.
+
+   .. versionadded:: 3.8
+
+.. data:: PyCF_ONLY_AST
+
+   Generates and returns an abstract syntax tree instead of returning
+   compiled code object.
```
Comment (pablogsal) [inline Doc/library/ast.rst]:
```suggestion
   Generates and returns an abstract syntax tree instead of returning a
   compiled code object.
```

**Comment by taleinat** [comment]
Ping, @pablogsal?

**Review by pablogsal — APPROVED** [review]

**Comment by rffontenelle** [comment]
@isidentical Hey there. I notice the new strings for translation in library/functions and one changed raised doubts. The current line (first of a long paragraph):

```
The optional argument *flags* and *dont_inherit* controls which
```

replaced this previous line:

```
The optional arguments *flags* and *dont_inherit* control which
```

Link: https://github.com/python/cpython/blame/master/Doc/library/functions.rst#L262

Was this plural-to-singular change intentional? If yes, would you please clarify?

**Comment by isidentical** [comment]
> Was this plural-to-singular change intentional?

Looks like an oversight. Would you mind sending a PR to fix that?

**Pull request merged**